### ISSUE TITLE
Understanding progress bar

### ISSUE THREAD

**Issue by Limtle** [PyTorchLightning/pytorch-lightning#4225]
When training or Validating on 2 nodes (8 gpus per node), Lightning show 16 same progress bars with different loss, such like
Epoch 0:   9%|▊         | 3/35 [00:14<02:29,  4.68s/it, loss=10.122, v_num=193413].
Epoch 0:   9%|▊         | 3/35 [00:14<02:29,  4.68s/it, loss=9.858, v_num=193413]
Epoch 0:   9%|▊         | 3/35 [00:14<02:29,  4.68s/it, loss=10.225, v_num=193413]
...
It means the output have 16 progress bars if training on 16 GPUs. I suppose that samples used for training are different on each GPU therefore leads to different progress bar. Moreover, the similar situation also show up in validation. I am wondering if different samples are distributed to each GPU during validation.
I use Dataset for training and IterableDataset for validation. 

```
Training: 0it [00:00, ?it/s]
Training:   0%|          | 0/35 [00:00<?, ?it/s]
Epoch 0:   0%|          | 0/35 [00:00<?, ?it/s] 
Epoch 0:   3%|▎         | 1/35 [00:08<04:50,  8.54s/it]
Epoch 0:   3%|▎         | 1/35 [00:08<04:50,  8.54s/it, loss=10.457, v_num=193413]
Epoch 0:   6%|▌         | 2/35 [00:11<03:06,  5.66s/it, loss=10.457, v_num=193413]
Epoch 0:   6%|▌         | 2/35 [00:11<03:06,  5.66s/it, loss=10.122, v_num=193413]
Epoch 0:   9%|▊         | 3/35 [00:14<02:29,  4.68s/it, loss=10.122, v_num=193413]
Epoch 0:   9%|▊         | 3/35 [00:14<02:29,  4.68s/it, loss=9.858, v_num=193413] 
Epoch 0:  11%|█▏        | 4/35 [00:16<02:09,  4.19s/it, loss=9.858, v_num=193413]
Epoch 0:  11%|█▏        | 4/35 [00:16<02:09,  4.19s/it, loss=9.626, v_num=193413]
Epoch 0:  14%|█▍        | 5/35 [00:19<01:57,  3.90s/it, loss=9.626, v_num=193413]
Epoch 0:  14%|█▍        | 5/35 [00:19<01:57,  3.90s/it, loss=9.432, v_num=193413]
Epoch 0:  17%|█▋        | 6/35 [00:22<01:47,  3.71s/it, loss=9.43
Training: 0it [00:00, ?it/s]
Training:   0%|          | 0/35 [00:00<?, ?it/s]
Epoch 0:   0%|          | 0/35 [00:00<?, ?it/s] 
Epoch 0:   3%|▎         | 1/35 [00:08<04:50,  8.54s/it]
Epoch 0:   3%|▎         | 1/35 [00:08<04:50,  8.54s/it, loss=10.468, v_num=193413]
Epoch 0:   6%|▌         | 2/35 [00:11<03:06,  5.66s/it, loss=10.468, v_num=193413]
Epoch 0:   6%|▌         | 2/35 [00:11<03:06,  5.66s/it, loss=10.122, v_num=193413]
Epoch 0:   9%|▊         | 3/35 [00:14<02:29,  4.68s/it, loss=10.122, v_num=193413]
Epoch 0:   9%|▊         | 3/35 [00:14<02:29,  4.68s/it, loss=9.846, v_num=193413] 
Epoch 0:  11%|█▏        | 4/35 [00:16<02:09,  4.19s/it, loss=9.846, v_num=193413]
Epoch 0:  11%|█▏        | 4/35 [00:16<02:09,  4.19s/it, loss=9.638, v_num=193413]
Epoch 0:  14%|█▍        | 5/35 [00:19<01:57,  3.90s/it, loss=9.638, v_num=193413]
Epoch 0:  14%|█▍        | 5/35 [00:19<01:57,  3.90s/it, loss=9.465, v_num=193413]
Epoch 0:  17%|█▋        | 6/35 [00:22<01:47,  3.71s/it, loss=9.46
Training: 0it [00:00, ?it/s]
Training:   0%|          | 0/35 [00:00<?, ?it/s]
Epoch 0:   0%|          | 0/35 [00:00<?, ?it/s] 
```
```
Validating: 0it [00:00, ?it/s][A

Validating: 0it [00:00, ?it/s][A

Validating: 0it [00:00, ?it/s][A

Validating: 0it [00:00, ?it/s][A

Validating: 0it [00:00, ?it/s][A

Validating: 0it [00:00, ?it/s][A

Validating: 0it [00:00, ?it/s][A

Validating:   0%|          | 1/30738 [00:01<16:11:25,  1.90s/it][A
Epoch 0: : 36it [02:15,  3.76s/it, loss=7.437, v_num=193413]                      

Validating:   0%|          | 1/30738 [00:01<16:08:01,  1.89s/it][A
Epoch 0: : 36it [02:15,  3.76s/it, loss=7.469, v_num=193413]                      

Validating:   0%|          | 1/30738 [00:01<16:08:42,  1.89s/it][A
Epoch 0: : 36it [02:15,  3.76s/it, loss=7.447, v_num=193413]                      

Validating:   0%|          | 1/30738 [00:01<16:08:19,  1.89s/it][A
Epoch 0: : 36it [02:15,  3.76s/it, loss=7.470, v_num=193413]                      

Validating:   0%|          | 1/30738 [00:01<16:12:26,  1.90s/it][A
Epoch 0: : 36it [02:15,  3.76s/it, loss=7.443, v_num=193413]                      

Validating:   0%|          | 1/30738 [00:01<16:15:56,  1.91s/it][A
Epoch 0: : 36it [02:15,  3.76s/it, loss=7.438, v_num=193413]                      

Validating:   0%|          | 1/30738 [00:01<16:14:04,  1.90s/it][A
Epoch 0: : 36it [02:15,  3.76s/it, loss=7.433, v_num=193413]                      

```
```
class Train_Dataset(Dataset):
    def __init__(self, filepath):
        self.examples=torch.load(filepath)
        self.len = len(self.examples)
    def __len__(self):
        return self.len
    def __getitem__(self, idx):
        return self.examples[idx]

class Val_Dataset(IterableDataset):
    def __init__(self, filename_list):
        self.filename_list = filename_list
        self.len = 0
        for file_path in tqdm(self.filename_list):
            self.len+=len(torch.load(file_path))

    def __len__(self):
        return self.len

    def __iter__(self):
        for file_path in self.filename_list:
            for x in torch.load(file_path):
                yield x
       
```

**Comment by Borda** [comment]
@Limtle could you please share a complete example to reproduce? Does it happen only on DDP? Are all 16 updated in parallel or just the last one?

**Comment by Limtle** [comment]
Sorry . the code can't be shared. But the following is my log file.
I will share a simple example as soon as possible.
```
GPU available: True, used: True
TPU available: False, using: 0 TPU cores
GPU available: True, used: True
TPU available: False, using: 0 TPU cores
Multi-processing is handled by Slurm.
Multi-processing is handled by Slurm.
LOCAL_RANK: 6 - CUDA_VISIBLE_DEVICES: [0,1,2,3,4,5,6,7]
LOCAL_RANK: 1 - CUDA_VISIBLE_DEVICES: [0,1,2,3,4,5,6,7]
Using native 16bit precision.
Using native 16bit precision.
initializing ddp: GLOBAL_RANK: 14, MEMBER: 15/16
initializing ddp: GLOBAL_RANK: 1, MEMBER: 2/16
GPU available: True, used: True
TPU available: False, using: 0 TPU cores
Multi-processing is handled by Slurm.
LOCAL_RANK: 2 - CUDA_VISIBLE_DEVICES: [0,1,2,3,4,5,6,7]
Using native 16bit precision.
GPU available: True, used: True
TPU available: False, using: 0 TPU cores
Multi-processing is handled by Slurm.
LOCAL_RANK: 3 - CUDA_VISIBLE_DEVICES: [0,1,2,3,4,5,6,7]
Using native 16bit precision.
initializing ddp: GLOBAL_RANK: 10, MEMBER: 11/16
GPU available: True, used: True
TPU available: False, using: 0 TPU cores
Multi-processing is handled by Slurm.
LOCAL_RANK: 7 - CUDA_VISIBLE_DEVICES: [0,1,2,3,4,5,6,7]
Using native 16bit precision.
initializing ddp: GLOBAL_RANK: 11, MEMBER: 12/16
initializing ddp: GLOBAL_RANK: 15, MEMBER: 16/16
GPU available: True, used: True
TPU available: False, using: 0 TPU cores
Multi-processing is handled by Slurm.
LOCAL_RANK: 7 - CUDA_VISIBLE_DEVICES: [0,1,2,3,4,5,6,7]
Using native 16bit precision.
initializing ddp: GLOBAL_RANK: 7, MEMBER: 8/16
GPU available: True, used: True
TPU available: False, using: 0 TPU cores
Multi-processing is handled by Slurm.
LOCAL_RANK: 5 - CUDA_VISIBLE_DEVICES: [0,1,2,3,4,5,6,7]
Using native 16bit precision.
initializing ddp: GLOBAL_RANK: 5, MEMBER: 6/16
GPU available: True, used: True
TPU available: False, using: 0 TPU cores
Multi-processing is handled by Slurm.
LOCAL_RANK: 3 - CUDA_VISIBLE_DEVICES: [0,1,2,3,4,5,6,7]
Using native 16bit precision.
GPU available: True, used: True
TPU available: False, using: 0 TPU cores
initializing ddp: GLOBAL_RANK: 3, MEMBER: 4/16
Multi-processing is handled by Slurm.
LOCAL_RANK: 6 - CUDA_VISIBLE_DEVICES: [0,1,2,3,4,5,6,7]
Using native 16bit precision.
initializing ddp: GLOBAL_RANK: 6, MEMBER: 7/16
GPU available: True, used: True
TPU available: False, using: 0 TPU cores
Multi-processing is handled by Slurm.
LOCAL_RANK: 4 - CUDA_VISIBLE_DEVICES: [0,1,2,3,4,5,6,7]
Using native 16bit precision.
initializing ddp: GLOBAL_RANK: 4, MEMBER: 5/16
GPU available: True, used: True
TPU available: False, using: 0 TPU cores
Multi-processing is handled by Slurm.
LOCAL_RANK: 2 - CUDA_VISIBLE_DEVICES: [0,1,2,3,4,5,6,7]
Using native 16bit precision.
initializing ddp: GLOBAL_RANK: 2, MEMBER: 3/16
GPU available: True, used: True
TPU available: False, using: 0 TPU cores
Multi-processing is handled by Slurm.
LOCAL_RANK: 5 - CUDA_VISIBLE_DEVICES: [0,1,2,3,4,5,6,7]
Using native 16bit precision.
initializing ddp: GLOBAL_RANK: 13, MEMBER: 14/16
GPU available: True, used: True
TPU available: False, using: 0 TPU cores
Multi-processing is handled by Slurm.
LOCAL_RANK: 4 - CUDA_VISIBLE_DEVICES: [0,1,2,3,4,5,6,7]
Using native 16bit precision.
initializing ddp: GLOBAL_RANK: 12, MEMBER: 13/16
GPU available: True, used: True
TPU available: False, using: 0 TPU cores
Multi-processing is handled by Slurm.
LOCAL_RANK: 1 - CUDA_VISIBLE_DEVICES: [0,1,2,3,4,5,6,7]
Using native 16bit precision.
initializing ddp: GLOBAL_RANK: 9, MEMBER: 10/16
GPU available: True, used: True
TPU available: False, using: 0 TPU cores
Multi-processing is handled by Slurm.
LOCAL_RANK: 0 - CUDA_VISIBLE_DEVICES: [0,1,2,3,4,5,6,7]
Using native 16bit precision.
initializing ddp: GLOBAL_RANK: 0, MEMBER: 1/16
GPU available: True, used: True
TPU available: False, using: 0 TPU cores
Multi-processing is handled by Slurm.
LOCAL_RANK: 0 - CUDA_VISIBLE_DEVICES: [0,1,2,3,4,5,6,7]
Using native 16bit precision.
initializing ddp: GLOBAL_RANK: 8, MEMBER: 9/16
read file : 0
read file : 0
read file : 0
read file : 0
read file : 0
read file : 0
read file : 0
read file : 0
read file : 0
read file : 0
read file : 0
read file : 0
read file : 0
read file : 0
read file : 0
read file : 0
Set SLURM handle signals.
Set SLURM handle signals.
Set SLURM handle signals.
Set SLURM handle signals.
Set SLURM handle signals.
Set SLURM handle signals.
Set SLURM handle signals.
Set SLURM handle signals.
Set SLURM handle signals.
Set SLURM handle signals.
Set SLURM handle signals.
Set SLURM handle signals.
Set SLURM handle signals.
Set SLURM handle signals.
Set SLURM handle signals.
Set SLURM handle signals.
get train_dataloader
read file : 0
get train_dataloader
get train_dataloader
get train_dataloader
read file : 0
get train_dataloader
get train_dataloader
get train_dataloader
get train_dataloader
get train_dataloader
get train_dataloader
read file : 0
get train_dataloader
read file : 0
get train_dataloader
get train_dataloader
read file : 0
get train_dataloader
get train_dataloader
read file : 0
read file : 0
read file : 0
read file : 0
read file : 0
read file : 0
get train_dataloader
read file : 0
read file : 0
read file : 0
read file : 0
read file : 0

Training: 0it [00:00, ?it/s]
Training:   0%|          | 0/18 [00:00<?, ?it/s]
Epoch 0:   0%|          | 0/18 [00:00<?, ?it/s] 
Epoch 0:   6%|▌         | 1/18 [00:08<02:29,  8.77s/it]
Epoch 0:   6%|▌         | 1/18 [00:08<02:29,  8.77s/it, loss=10.478, v_num=193315]
Epoch 0:  11%|█         | 2/18 [00:11<01:33,  5.82s/it, loss=10.478, v_num=193315]
Epoch 0:  11%|█         | 2/18 [00:11<01:33,  5.82s/it, loss=10.227, v_num=193315]
Epoch 0:  17%|█▋        | 3/18 [00:14<01:13,  4.87s/it, loss=10.227, v_num=193315]
Epoch 0:  17%|█▋        | 3/18 [00:14<01:13,  4.87s/it, loss=10.072, v_num=193315]
Epoch 0:  22%|██▏       | 4/18 [00:17<01:02,  4.47s/it, loss=10.072, v_num=193315]
Epoch 0:  22%|██▏       | 4/18 [00:17<01:02,  4.47s/it, loss=9.947, v_num=193315] 
Epoch 0:  28%|██▊       | 5/18 [00:20<00:53,  4.14s/it, loss=9.947, v_num=193315]
Epoch 0:  28%|██▊       | 5/18 [00:20<00:53,  4.14s/it, loss=9.850, v_num=193315]
Epoch 0:  33%|███▎      | 6/18 [00:23<00:47,  3
Training: 0it [00:00, ?it/s]
Training:   0%|          | 0/18 [00:00<?, ?it/s]
Epoch 0:   0%|          | 0/18 [00:00<?, ?it/s] 
Epoch 0:   6%|▌         | 1/18 [00:08<02:29,  8.77s/it]
Epoch 0:   6%|▌         | 1/18 [00:08<02:29,  8.77s/it, loss=10.473, v_num=193315]
Epoch 0:  11%|█         | 2/18 [00:11<01:33,  5.82s/it, loss=10.473, v_num=193315]
Epoch 0:  11%|█         | 2/18 [00:11<01:33,  5.82s/it, loss=10.224, v_num=193315]
Epoch 0:  17%|█▋        | 3/18 [00:14<01:13,  4.87s/it, loss=10.224, v_num=193315]
Epoch 0:  17%|█▋        | 3/18 [00:14<01:13,  4.87s/it, loss=10.047, v_num=193315]
Epoch 0:  22%|██▏       | 4/18 [00:17<01:02,  4.47s/it, loss=10.047, v_num=193315]
Epoch 0:  22%|██▏       | 4/18 [00:17<01:02,  4.47s/it, loss=9.926, v_num=193315] 
Epoch 0:  28%|██▊       | 5/18 [00:20<00:53,  4.14s/it, loss=9.926, v_num=193315]
Epoch 0:  28%|██▊       | 5/18 [00:20<00:53,  4.14s/it, loss=9.847, v_num=193315]
Epoch 0:  33%|███▎      | 6/18 [00:23<00:47,  3
Training: 0it [00:00, ?it/s]
Training:   0%|          | 0/18 [00:00<?, ?it/s]
Epoch 0:   0%|          | 0/18 [00:00<?, ?it/s] 
Epoch 0:   6%|▌         | 1/18 [00:08<02:29,  8.77s/it]
Epoch 0:   6%|▌         | 1/18 [00:08<02:29,  8.77s/it, loss=10.502, v_num=193315]
Epoch 0:  11%|█         | 2/18 [00:11<01:33,  5.82s/it, loss=10.502, v_num=193315]
Epoch 0:  11%|█         | 2/18 [00:11<01:33,  5.82s/it, loss=10.244, v_num=193315]
Epoch 0:  17%|█▋        | 3/18 [00:14<01:13,  4.87s/it, loss=10.244, v_num=193315]
Epoch 0:  17%|█▋        | 3/18 [00:14<01:13,  4.87s/it, loss=10.073, v_num=193315]
Epoch 0:  22%|██▏       | 4/18 [00:17<01:02,  4.47s/it, loss=10.073, v_num=193315]
Epoch 0:  22%|██▏       | 4/18 [00:17<01:02,  4.47s/it, loss=9.958, v_num=193315] 
Epoch 0:  28%|██▊       | 5/18 [00:20<00:53,  4.14s/it, loss=9.958, v_num=193315]
Epoch 0:  28%|██▊       | 5/18 [00:20<00:53,  4.14s/it, loss=9.854, v_num=193315]
Epoch 0:  33%|███▎      | 6/18 [00:23<00:47,  3
Training: 0it [00:00, ?it/s]
Training:   0%|          | 0/18 [00:00<?, ?it/s]
Epoch 0:   0%|          | 0/18 [00:00<?, ?it/s] 
Epoch 0:   6%|▌         | 1/18 [00:08<02:29,  8.77s/it]
Epoch 0:   6%|▌         | 1/18 [00:08<02:29,  8.77s/it, loss=10.478, v_num=193315]
Epoch 0:  11%|█         | 2/18 [00:11<01:33,  5.82s/it, loss=10.478, v_num=193315]
Epoch 0:  11%|█         | 2/18 [00:11<01:33,  5.82s/it, loss=10.221, v_num=193315]
Epoch 0:  17%|█▋        | 3/18 [00:14<01:13,  4.87s/it, loss=10.221, v_num=193315]
Epoch 0:  17%|█▋        | 3/18 [00:14<01:13,  4.87s/it, loss=10.049, v_num=193315]
Epoch 0:  22%|██▏       | 4/18 [00:17<01:02,  4.47s/it, loss=10.049, v_num=193315]
Epoch 0:  22%|██▏       | 4/18 [00:17<01:02,  4.47s/it, loss=9.914, v_num=193315] 
Epoch 0:  28%|██▊       | 5/18 [00:20<00:53,  4.14s/it, loss=9.914, v_num=193315]
Epoch 0:  28%|██▊       | 5/18 [00:20<00:53,  4.14s/it, loss=9.835, v_num=193315]
Epoch 0:  33%|███▎      | 6/18 [00:23<00:47,  3
Training: 0it [00:00, ?it/s]
Training:   0%|          | 0/18 [00:00<?, ?it/s]
Epoch 0:   0%|          | 0/18 [00:00<?, ?it/s] 
Epoch 0:   6%|▌         | 1/18 [00:08<02:29,  8.77s/it]
Epoch 0:   6%|▌         | 1/18 [00:08<02:29,  8.77s/it, loss=10.493, v_num=193315]
Epoch 0:  11%|█         | 2/18 [00:11<01:33,  5.82s/it, loss=10.493, v_num=193315]
Epoch 0:  11%|█         | 2/18 [00:11<01:33,  5.82s/it, loss=10.223, v_num=193315]
Epoch 0:  17%|█▋        | 3/18 [00:14<01:13,  4.87s/it, loss=10.223, v_num=193315]
Epoch 0:  17%|█▋        | 3/18 [00:14<01:13,  4.87s/it, loss=10.060, v_num=193315]
Epoch 0:  22%|██▏       | 4/18 [00:17<01:02,  4.47s/it, loss=10.060, v_num=193315]
Epoch 0:  22%|██▏       | 4/18 [00:17<01:02,  4.47s/it, loss=9.923, v_num=193315] 
Epoch 0:  28%|██▊       | 5/18 [00:20<00:53,  4.14s/it, loss=9.923, v_num=193315]
Epoch 0:  28%|██▊       | 5/18 [00:20<00:53,  4.14s/it, loss=9.833, v_num=193315]
Epoch 0:  33%|███▎      | 6/18 [00:23<00:47,  3
Training: 0it [00:00, ?it/s]
Training:   0%|          | 0/18 [00:00<?, ?it/s]
Epoch 0:   0%|          | 0/18 [00:00<?, ?it/s] 
Epoch 0:   6%|▌         | 1/18 [00:08<02:29,  8.77s/it]
Epoch 0:   6%|▌         | 1/18 [00:08<02:29,  8.77s/it, loss=10.496, v_num=193315]
Epoch 0:  11%|█         | 2/18 [00:11<01:33,  5.82s/it, loss=10.496, v_num=193315]
Epoch 0:  11%|█         | 2/18 [00:11<01:33,  5.82s/it, loss=10.218, v_num=193315]
Epoch 0:  17%|█▋        | 3/18 [00:14<01:13,  4.87s/it, loss=10.218, v_num=193315]
Epoch 0:  17%|█▋        | 3/18 [00:14<01:13,  4.87s/it, loss=10.035, v_num=193315]
Epoch 0:  22%|██▏       | 4/18 [00:17<01:02,  4.47s/it, loss=10.035, v_num=193315]
Epoch 0:  22%|██▏       | 4/18 [00:17<01:02,  4.47s/it, loss=9.917, v_num=193315] 
Epoch 0:  28%|██▊       | 5/18 [00:20<00:53,  4.14s/it, loss=9.917, v_num=193315]
Epoch 0:  28%|██▊       | 5/18 [00:20<00:53,  4.14s/it, loss=9.831, v_num=193315]
Epoch 0:  33%|███▎      | 6/18 [00:23<00:47,  3
Training: 0it [00:00, ?it/s]
Training:   0%|          | 0/18 [00:00<?, ?it/s]
Epoch 0:   0%|          | 0/18 [00:00<?, ?it/s] 
Epoch 0:   6%|▌         | 1/18 [00:08<02:29,  8.77s/it]
Epoch 0:   6%|▌         | 1/18 [00:08<02:29,  8.77s/it, loss=10.505, v_num=193315]
Epoch 0:  11%|█         | 2/18 [00:11<01:33,  5.82s/it, loss=10.505, v_num=193315]
Epoch 0:  11%|█         | 2/18 [00:11<01:33,  5.82s/it, loss=10.231, v_num=193315]
Epoch 0:  17%|█▋        | 3/18 [00:14<01:13,  4.87s/it, loss=10.231, v_num=193315]
Epoch 0:  17%|█▋        | 3/18 [00:14<01:13,  4.87s/it, loss=10.049, v_num=193315]
Epoch 0:  22%|██▏       | 4/18 [00:17<01:02,  4.47s/it, loss=10.049, v_num=193315]
Epoch 0:  22%|██▏       | 4/18 [00:17<01:02,  4.47s/it, loss=9.908, v_num=193315] 
Epoch 0:  28%|██▊       | 5/18 [00:20<00:53,  4.14s/it, loss=9.908, v_num=193315]
Epoch 0:  28%|██▊       | 5/18 [00:20<00:53,  4.14s/it, loss=9.822, v_num=193315]
Epoch 0:  33%|███▎      | 6/18 [00:23<00:47,  3
Training: 0it [00:00, ?it/s]
Training:   0%|          | 0/18 [00:00<?, ?it/s]
Epoch 0:   0%|          | 0/18 [00:00<?, ?it/s] 
Epoch 0:   6%|▌         | 1/18 [00:08<02:29,  8.77s/it]
Epoch 0:   6%|▌         | 1/18 [00:08<02:29,  8.77s/it, loss=10.500, v_num=193315]
Epoch 0:  11%|█         | 2/18 [00:11<01:33,  5.82s/it, loss=10.500, v_num=193315]
Epoch 0:  11%|█         | 2/18 [00:11<01:33,  5.82s/it, loss=10.201, v_num=193315]
Epoch 0:  17%|█▋        | 3/18 [00:14<01:13,  4.87s/it, loss=10.201, v_num=193315]
Epoch 0:  17%|█▋        | 3/18 [00:14<01:13,  4.87s/it, loss=10.035, v_num=193315]
Epoch 0:  22%|██▏       | 4/18 [00:17<01:02,  4.47s/it, loss=10.035, v_num=193315]
Epoch 0:  22%|██▏       | 4/18 [00:17<01:02,  4.47s/it, loss=9.914, v_num=193315] 
Epoch 0:  28%|██▊       | 5/18 [00:20<00:53,  4.14s/it, loss=9.914, v_num=193315]
Epoch 0:  28%|██▊       | 5/18 [00:20<00:53,  4.14s/it, loss=9.813, v_num=193315]
Epoch 0:  33%|███▎      | 6/18 [00:23<00:47,  3
Training: 0it [00:00, ?it/s]
Training:   0%|          | 0/18 [00:00<?, ?it/s]
Epoch 0:   0%|          | 0/18 [00:00<?, ?it/s] 
Epoch 0:   6%|▌         | 1/18 [00:08<02:29,  8.77s/it]
Epoch 0:   6%|▌         | 1/18 [00:08<02:29,  8.77s/it, loss=10.480, v_num=193315]
Epoch 0:  11%|█         | 2/18 [00:11<01:33,  5.82s/it, loss=10.480, v_num=193315]
Epoch 0:  11%|█         | 2/18 [00:11<01:33,  5.82s/it, loss=10.213, v_num=193315]
Epoch 0:  17%|█▋        | 3/18 [00:14<01:13,  4.87s/it, loss=10.213, v_num=193315]
Epoch 0:  17%|█▋        | 3/18 [00:14<01:13,  4.87s/it, loss=10.042, v_num=193315]
Epoch 0:  22%|██▏       | 4/18 [00:17<01:02,  4.47s/it, loss=10.042, v_num=193315]
Epoch 0:  22%|██▏       | 4/18 [00:17<01:02,  4.47s/it, loss=9.911, v_num=193315] 
Epoch 0:  28%|██▊       | 5/18 [00:20<00:53,  4.14s/it, loss=9.911, v_num=193315]
Epoch 0:  28%|██▊       | 5/18 [00:20<00:53,  4.14s/it, loss=9.828, v_num=193315]
Epoch 0:  33%|███▎      | 6/18 [00:23<00:47,  3
Training: 0it [00:00, ?it/s]
Training:   0%|          | 0/18 [00:00<?, ?it/s]
Epoch 0:   0%|          | 0/18 [00:00<?, ?it/s] 
Epoch 0:   6%|▌         | 1/18 [00:08<02:29,  8.77s/it]
Epoch 0:   6%|▌         | 1/18 [00:08<02:29,  8.77s/it, loss=10.498, v_num=193315]
Epoch 0:  11%|█         | 2/18 [00:11<01:33,  5.82s/it, loss=10.498, v_num=193315]
Epoch 0:  11%|█         | 2/18 [00:11<01:33,  5.82s/it, loss=10.233, v_num=193315]
Epoch 0:  17%|█▋        | 3/18 [00:14<01:13,  4.87s/it, loss=10.233, v_num=193315]
Epoch 0:  17%|█▋        | 3/18 [00:14<01:13,  4.87s/it, loss=10.059, v_num=193315]
Epoch 0:  22%|██▏       | 4/18 [00:17<01:02,  4.47s/it, loss=10.059, v_num=193315]
Epoch 0:  22%|██▏       | 4/18 [00:17<01:02,  4.47s/it, loss=9.941, v_num=193315] 
Epoch 0:  28%|██▊       | 5/18 [00:20<00:53,  4.14s/it, loss=9.941, v_num=193315]
Epoch 0:  28%|██▊       | 5/18 [00:20<00:53,  4.14s/it, loss=9.845, v_num=193315]
Epoch 0:  33%|███▎      | 6/18 [00:23<00:47,  3
Training: 0it [00:00, ?it/s]
Training:   0%|          | 0/18 [00:00<?, ?it/s]
Epoch 0:   0%|          | 0/18 [00:00<?, ?it/s] 
Epoch 0:   6%|▌         | 1/18 [00:08<02:29,  8.77s/it]
Epoch 0:   6%|▌         | 1/18 [00:08<02:29,  8.77s/it, loss=10.482, v_num=193315]
Epoch 0:  11%|█         | 2/18 [00:11<01:33,  5.82s/it, loss=10.482, v_num=193315]
Epoch 0:  11%|█         | 2/18 [00:11<01:33,  5.82s/it, loss=10.222, v_num=193315]
Epoch 0:  17%|█▋        | 3/18 [00:14<01:13,  4.87s/it, loss=10.222, v_num=193315]
Epoch 0:  17%|█▋        | 3/18 [00:14<01:13,  4.87s/it, loss=10.047, v_num=193315]
Epoch 0:  22%|██▏       | 4/18 [00:17<01:02,  4.47s/it, loss=10.047, v_num=193315]
Epoch 0:  22%|██▏       | 4/18 [00:17<01:02,  4.47s/it, loss=9.923, v_num=193315] 
Epoch 0:  28%|██▊       | 5/18 [00:20<00:53,  4.14s/it, loss=9.923, v_num=193315]
Epoch 0:  28%|██▊       | 5/18 [00:20<00:53,  4.14s/it, loss=9.823, v_num=193315]
Epoch 0:  33%|███▎      | 6/18 [00:23<00:47,  3
.....
```

**Comment by Limtle** [comment]
I have another example from [pytorch_lightning](https://colab.research.google.com/github/PytorchLightning/pytorch-lightning/blob/master/notebooks/03-basic-gan.ipynb#scrollTo=0E2QDjl5yWtz) . use 1 node (8 gpus) also have same `situation.
```
```

**Comment by Limtle** [comment]
@Borda How to check all 16 updated in parallel or just the last one?

**Comment by Limtle** [comment]
closed  #4437
